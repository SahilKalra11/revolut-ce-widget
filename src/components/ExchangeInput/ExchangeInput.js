import React from "react";
import "./ExchangeInput.scss";
import { formatToTwoDigitsNumber, getCurrencySymbol } from "../../helpers";

const ExchangeInput = ({
  value,
  onChange,
  cardId,
  rates,
  currency,
  currencyFrom,
  focus,
}) => {
  const rate = (1 / rates[currency]).toFixed(2);
  let course =
    cardId === "bottom-card"
      ? `${getCurrencySymbol(currency)}1 = ${getCurrencySymbol(
          currencyFrom
        )}${rate}`
      : "";

  return (
    <section className="exchange__action">
      <input
        autoFocus={focus}
        className="exchange__input"
        type="text"
        value={value === "" ? "" : formatToTwoDigitsNumber(value)}
        onChange={onChange}
      />

      <div className="exchange__course">{course}</div>
    </section>
  );
};

export default ExchangeInput;
