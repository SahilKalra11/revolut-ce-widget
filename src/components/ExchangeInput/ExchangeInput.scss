@import '../../utilities/variables';
@import '../../utilities/mixins';

.exchange__input {
  @include currency-style;
  width: 100%;
  background: transparent;
  text-align: right;
}

.exchange__action {
  flex: 50%;
  text-align: right;
}
