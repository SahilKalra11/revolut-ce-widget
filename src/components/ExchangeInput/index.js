export { default } from './ExchangeInput.js';