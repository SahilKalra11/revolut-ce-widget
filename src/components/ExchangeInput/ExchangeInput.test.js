import React from "react";
import ReactDOM from "react-dom";
import render from "react-test-renderer";
import ExchangeInput from "./ExchangeInput";
import { unmountComponentAtNode } from "react-dom";

describe("ExchangeInput", () => {
  let container = null;
  beforeEach(() => {
    container = document.createElement("div");
    document.body.appendChild(container);
  });

  afterEach(() => {
    unmountComponentAtNode(container);
    container.remove();
    container = null;
  });

  const props = {
    value: 10,
    cardId: "cardId",
    currency: "USD",
    currencyFrom: "USD",
    rates: {
      USD: 1,
      EUR: 1.2,
      GBP: 1.3,
    },
    onChange: jest.fn(),
  };

  it("renders component without crashing", () => {
    ReactDOM.render(<ExchangeInput {...props} />, container);
  });

  test("matches to snapshot", () => {
    const component = render.create(<ExchangeInput {...props} />);
    let componentTree = component.toJSON();
    expect(componentTree).toMatchSnapshot();
  });
});
