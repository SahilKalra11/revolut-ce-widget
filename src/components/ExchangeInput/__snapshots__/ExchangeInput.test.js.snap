// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExchangeInput matches to snapshot 1`] = `
<section
  className="exchange__action"
>
  <input
    className="exchange__input"
    onChange={[MockFunction]}
    type="text"
    value={10}
  />
  <div
    className="exchange__course"
  >
    
  </div>
</section>
`;
